src/playground/App.js

```diff
@@ -6,41 +6,44 @@ import CrashAlert from "./components/CrashAlert";
 import Footer from "./components/Footer";
 import CodeEditor from "./components/CodeEditor";
 import { Linter, SourceCodeFixer } from "./node_modules/eslint/lib/linter/";
-import { Legacy } from "@eslint/eslintrc/universal";
 import Unicode from "./utils/unicode";
 import Configuration from "./components/Configuration";
 import Split from "react-split";
 import debounce from "./utils/debounce";
 import "./scss/split-pane.scss";
 
-const linter = new Linter();
-const rules = linter.getRules();
+const linter = new Linter({ configType: "flat" });
+const legacyLinter = new Linter({ configType: "eslintrc" });
+const rules = legacyLinter.getRules();
 const ruleNames = Array.from(rules.keys());
 const rulesMeta = Array.from(rules.entries()).reduce((result, [key, value]) => {
     result[key] = value.meta;
     return result;
 }, {});
 
-const defaultParserOptions = {
-    ecmaFeatures: {},
-    ecmaVersion: "latest",
-    sourceType: "script"
+const defaultLanguageOptions = {
+    parserOptions: {
+        ecmaFeatures: {}
+    }
 };
 
+const isEmptyObject = obj => obj && Object.keys(obj).length === 0 && obj.constructor === Object;
+
 const fillOptionsDefaults = options =>
-    (options
+    (!isEmptyObject(options)
         ? {
-            env: {},
             rules: {},
             ...options,
-            parserOptions: {
-                ...defaultParserOptions,
-                ...options.parserOptions
+            languageOptions: {
+                ...options.languageOptions,
+                parserOptions: {
+                    ecmaFeatures: {},
+                    ...options.languageOptions?.parserOptions
```

Comment:
it will throw an error when `options.languageOptions` is `null/undef`?

Reply:
`{ ...undefined }` doesn't throw, it's just a no-op.